### PR TITLE
fix: Remove unary operators

### PR DESCRIPTION
**Pull request by teolemon**
### Why
- Because the unary ++ and -- operators are subject to automatic semicolon insertion, differences in whitespace can change semantics of source code.

### CODE REVIEW

File: html/js/folksonomy.js [hunk 1]
```diff
@@ -315,7 +315,7 @@ function displayFolksonomyPropertyValues() {
         let content = "";
         // Sort by property
         const d = data.sort(function(a,b){ return a.k <b.k ? 1 :-1; });
-        while (index--) {
+        while (index-=) {
```
Comment (alexgarel) [inline html/js/folksonomy.js:318]:
```suggestion
        while (index-=1) {
```

File: html/js/folksonomy.js [hunk 2]
```diff
@@ -374,7 +374,7 @@ function displayProductsWithProperty(_property, _value) {
             '<th class="values">Values</th>' +
             '</tr>' +
             '<tbody id="free_prop_body">';
-        while (index--) {
+        while (index-=) {
```
Comment (alexgarel) [inline html/js/folksonomy.js:377]:
```suggestion
        while (index-=1) {
```

File: html/js/folksonomy.js [hunk 3]
```diff
@@ -687,7 +687,7 @@ function displayFolksonomyForm() {
         let index = data.length;
         let content = "";
         const d = data.sort(function(a,b){ return a.k <b.k ?1 :-1; });
-        while (index--) {
+        while (index-=) {
```
Comment (alexgarel) [inline html/js/folksonomy.js:690]:
```suggestion
        while (index-=1) {
```

**Review by alexgarel — CHANGES_REQUESTED** [review]
I think you forgot the 1

**Comment by VaiTon** [comment]
NPM test is failing because of:

```
 /home/runner/work/openfoodfacts-server/openfoodfacts-server/html/js/folksonomy.js
Error:   318:16  error    Expected a conditional expression and instead saw an assignment  no-cond-assign
Error:   377:16  error    Expected a conditional expression and instead saw an assignment  no-cond-assign
Error:   425:16  error    Expected a conditional expression and instead saw an assignment  no-cond-assign
Error:   690:16  error    Expected a conditional expression and instead saw an assignment  no-cond-assign
Warning:   991:13  warning  Unary operator '++' used                                         no-plusplus
```

**Comment by alexgarel** [comment]
@VaiTon error is related to the PR (you see it in the *Files changed* tab).

Indeed the linter does not like this way of writing. So I think the `index--` was the right way of writing…

@teolemon I would suggest closing this.

**Comment by teolemon** [comment]
ok

**Pull request closed without merging**